### PR TITLE
Adds execute script API

### PR DESCRIPTION
**Pull request by kolchfa-aws**
Adds execute script API

### Checklist
- [x ] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by alicejw-aws — APPROVED** [review]
Looks great!

**Review by natebower — COMMENTED** [review]
@kolchfa-aws Please see my comments and changes and let me know if you have any questions. Thanks!

File: _api-reference/script-apis/exec-script.md [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+
+### Painless test context
+
+The `painless_test` context is the default script context that provides only the `params` variable to the script. The returned result is always converted to a string. See the sample request above for a usage example.
+
+### Filter context
+
+The `filter` context runs the script as if the script was inside a script query. You must provide a test document in the context. The `_source`, stored fields and `_doc` variables will be available to the script.
+
+You can specify the following parameters to the filter context in the `context_setup`.
```
[the remark this answers is not in the document]
Comment (kolchfa-aws) [reply _api-reference/script-apis/exec-script.md:73]: We can change to "for".

File: _api-reference/script-apis/exec-script.md [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+    "index": "testindex1",
+    "document": {
+      "grad": true,
+      "gpa": 3.79
+    }
+  }
+}
+```
+
+The response contains the result:
```
[the remark this answers is not in the document]
Comment (kolchfa-aws) [reply _api-reference/script-apis/exec-script.md:120]: Not really. `result` refers to the result of executing the script. So, what I mean is we calculate the result and return it in the response.

File: _api-reference/script-apis/exec-script.md [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+{
+  "result" : true
+}
+```
+
+### Score context
+
+The `score` context runs a script as if the script was in a `script_score` function in a `function_score` query.
+
+You can specify the following parameters to the score context in the `context_setup`.
```
[the remark this answers is not in the document]
Comment (kolchfa-aws) [reply _api-reference/script-apis/exec-script.md:132]: Changed to "for"

**Pull request merged**